packages/worker/native/src/lib.rs

```diff
@@ -36,92 +43,82 @@ impl Worker {
 
         Worker {
             core: Box::new(core),
-            condition: Condvar::new(),
-            suspended: Mutex::new(false),
+            sender,
         }
     }
 
     pub fn poll(&self, queue_name: String) -> ::temporal_sdk_core::Result<coresdk::Task> {
-        let _guard = self
-            .condition
-            .wait_while(self.suspended.lock().unwrap(), |suspended| *suspended)
-            .unwrap();
         self.core.poll_task(&queue_name)
     }
-
-    pub fn is_suspended(&self) -> bool {
-        *self.suspended.lock().unwrap()
-    }
-
-    pub fn suspend_polling(&self) {
-        *self.suspended.lock().unwrap() = true;
-        self.condition.notify_one();
-    }
-
-    pub fn resume_polling(&self) {
-        *self.suspended.lock().unwrap() = false;
-        self.condition.notify_one();
-    }
 }
 
 fn worker_new(mut cx: FunctionContext) -> JsResult<BoxedWorker> {
-    let worker = Arc::new(Worker::new());
-
-    Ok(cx.boxed(worker))
-}
-
-fn worker_poll(mut cx: FunctionContext) -> JsResult<JsUndefined> {
-    let worker = cx.argument::<BoxedWorker>(0)?;
-    let queue_name = cx.argument::<JsString>(1)?.value(&mut cx);
-    let callback = cx.argument::<JsFunction>(2)?.root(&mut cx);
-    let arc_worker = Arc::clone(&**worker); // deref Handle and JsBox
-    let arc_callback = Arc::new(callback);
+    // TODO: is capacity of 1 enough here?
```

Comment:
As long as you never ask the worker to poll without handling the result of the last poll, which makes sense in a single threaded context, so keeping the bound is probably a nice check.